cmd/tether/attach_test.go

```diff
@@ -0,0 +1,613 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io"
+	"net"
+	"sync"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/net/context"
+
+	"github.com/vmware/vic/metadata"
+)
+
+/////////////////////////////////////////////////////////////////////////////////////
+// TestAttachConfig sets up the config for attach testing - the grep will echo anything
+// sent and adds colour which is useful for tty testing
+//
+type TestAttachConfig struct{}
+
+func (c *TestAttachConfig) StoreConfig(*metadata.ExecutorConfig) (string, error) {
+	return "", errors.New("not implemented")
+}
+func (c *TestAttachConfig) LoadConfig() (*metadata.ExecutorConfig, error) {
+	config := metadata.ExecutorConfig{}
+
+	config.ID = "attach"
+	config.Name = "tether_test_executor"
+	config.Sessions = map[string]metadata.SessionConfig{
+		"attach": metadata.SessionConfig{
+			Common: metadata.Common{
+				ID:   "attach",
+				Name: "tether_test_session",
+			},
+			Tty:    false,
+			Attach: true,
+			Cmd: metadata.Cmd{
+				Path: "/usr/bin/tee",
+				// grep, matching everything, reading from stdin
+				Args: []string{"/usr/bin/tee", pathPrefix + "/tee.out"},
+				Env:  []string{},
+				Dir:  "/",
+			},
+		},
+	}
+
+	// generate a host key for the tether
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2014)
```

Comment:
Yeah, that's a typo which we fixed for spec package https://github.com/vmware/vic/blob/master/pkg/vsphere/spec/spec.go#L94